### DISCUSSION TITLE
Frigate crashes on restarting machine - fails to find TPU.

### DISCUSSION THREAD

**Discussion by utkarshjain00** [blakeblackshear/frigate#1194] [Ask A Question]
Everything worked well, then I restarted my Debian box. Frigate crashed on restart (logs below)

If I do: docker restart frigate, it comes back up like nothing was ever wrong. What can be done to make Frigate robust on restart?

Logs:

Process detector:coral:
frigate.edgetpu                INFO    : No EdgeTPU detected.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 152, in load_delegate
    delegate = Delegate(library, options)
  File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 111, in __init__
    raise ValueError(capture.message)
ValueError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/frigate/frigate/edgetpu.py", line 124, in run_detector
    object_detector = LocalObjectDetector(tf_device=tf_device, num_threads=num_threads)
  File "/opt/frigate/frigate/edgetpu.py", line 63, in __init__
    edge_tpu_delegate = load_delegate('libedgetpu.so.1.0', device_config)
  File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 154, in load_delegate
    raise ValueError('Failed to load delegate from {}\n{}'.format(
ValueError: Failed to load delegate from libedgetpu.so.1.0

frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
frigate.app                    INFO    : Stopping...

**Comment by blakeblackshear** [comment]
Do you have docker configured to restart frigate on exit?

**Reply by utkarshjain00** [reply]
It is set to unless-stopped, I guess I can try to switch it to on-failure. Not sure if that can create a restart loop. 

In general, I see that the container exits on many error conditions, like say MQTT server not available, rather than let the WebUI work and connect MQTT when available.

**Reply by blakeblackshear** [reply]
It exits on failures by design. Frigate's performance is in large part because of **lots** of multiprocessing. Coordinating across all those processes is very difficult and error prone, so the best way to "try again" in almost every scenario is to just exit the container and let docker/kubernetes/supervisor restart it.

**Reply by utkarshjain00** [reply]
Ok, got it. Confirmed that with following policy in docker-compose, the container works on machine restarts smoothly. Thanks for quick responses and all the great work on the project.

restart: on-failure
